tests/test_vertical_portscans.py

```diff
@@ -208,3 +227,307 @@ def test_check_if_enough_dports_to_trigger_an_evidence(
+    vertical_ps.combine_evidence()
+    assert mock_set_evidence.call_count == 2
+
+
+def test_decide_if_time_to_set_evidence_or_combine_new_cache_key(mock_db):
+    vertical_ps = ModuleFactory().create_vertical_portscan_obj(mock_db)
+    evidence = {
+        'timestamp': 1700828217.314165,
+        'pkts_sent': 100,
+        'protocol': 'TCP',
+        'profileid': 'profile_1.1.1.1',
+        'twid': 'timewindow0',
+        'uid': ['uid1', 'uid2'],
+        'amount_of_dports': 10,
+        'dstip': '8.8.8.8',
+        'state': 'Not Established',
+    }
+    cache_key = 'new_cache_key'
+
+    mock_set_evidence = mock_db.set_evidence
+
+    enough = vertical_ps.decide_if_time_to_set_evidence_or_combine(evidence, cache_key)
+
+    assert enough
+    assert vertical_ps.alerted_once_vertical_ps[cache_key]
+    mock_set_evidence.assert_called_once()
+    
+def test_check_no_connections(mock_db):
+    vertical_ps = ModuleFactory().create_vertical_portscan_obj(mock_db)
+    profileid = "profile_1.1.1.1"
+    twid = "timewindow0"
+
+    mock_db.get_data_from_profile_tw.return_value = {}
+    mock_set_evidence = mock_db.set_evidence
+
+    vertical_ps.check(profileid, twid)
+
+    mock_set_evidence.assert_not_called()
+    
+def test_get_not_established_dst_ips_invalid_protocol(mock_db):
+    vertical_ps = ModuleFactory().create_vertical_portscan_obj(mock_db)
+    profileid = "profile_1.1.1.1"
+    twid = "timewindow0"
+    protocol = "INVALID"
+    state = "Not Established"
+    role = "Client"
+    type_data = "IPs"
+    direction = "Dst"
+
+   
+    mock_db.get_data_from_profile_tw.return_value = {}
+
+    dstips = vertical_ps.get_not_established_dst_ips(protocol, state, profileid, twid)
+
+    assert dstips == {}     
+    
+def test_check_if_enough_dports_to_trigger_an_evidence_large_values(mock_db):
+    vertical_ps = ModuleFactory().create_vertical_portscan_obj(mock_db)
+    profileid = 'profile_1.1.1.1'
+    timewindow = 'timewindow0'
+    dstip = '8.8.8.8'
+    cache_key = vertical_ps.get_cache_key(profileid, timewindow, dstip)
+
+    
+    amount_of_dports = 65536
+    enough = vertical_ps.check_if_enough_dports_to_trigger_an_evidence(cache_key, amount_of_dports)
+    assert enough
+
+   
+    amount_of_dports = 1
+    enough = vertical_ps.check_if_enough_dports_to_trigger_an_evidence(cache_key, amount_of_dports)
+    assert not enough    
+  
+
+def test_get_not_established_dst_ips_db_exception(mock_db):
+    vertical_ps = ModuleFactory().create_vertical_portscan_obj(mock_db)
+    protocol = "TCP"
+    state = "Not Established"
+    profileid = "profile_1.1.1.1"
+    twid = "timewindow0"
+
+    mock_db.get_data_from_profile_tw.side_effect = Exception("Database error")
+
+    with pytest.raises(Exception):
+        vertical_ps.get_not_established_dst_ips(protocol, state, profileid, twid)
+        
+def test_check_overlapping_dstports(mock_db):
+    vertical_ps = ModuleFactory().create_vertical_portscan_obj(mock_db)
```

Comment:
Hey @Sekhar-Kumar-Dash I'm having trouble understanding this test
why are you testing overlapping ports? what can go wrong here?
Also i recommend using function docstrings so it's easier to know the answer to the above questions without having to ask you every time :)

Reply:
hey @AlyaGomaa The i have included this test is to ensure that the VerticalPortscan module correctly aggregates the packet counts for overlapping destination ports when scanning both TCP and UDP protocols.  it is not a critical test we can remove it if you dont want it 

am i getting it wrong??
Also thanks for the docstrings suggestion   i will keep that in mind for future unittests PRs